src/data/index.ts

```diff
@@ -22,11 +22,18 @@ export function getTokenIdentifier(token: Token): string {
  * @returns Returns `${tokenCode}:${issuerKey}`.
  */
 export function getBalanceIdentifier(balance: Horizon.BalanceLine): string {
-  if (balance.asset_type === "native" || !balance.asset_issuer) {
-    return "native";
+  const NATIVE_TYPE = "native";
+  if ("asset_issuer" in balance) {
+    if (!balance.asset_issuer) {
+      return NATIVE_TYPE;
+    }
+    return `${balance.asset_code}:${balance.asset_issuer}`;
+  }
+  if (balance.asset_type === NATIVE_TYPE) {
+    return NATIVE_TYPE;
   }
 
-  return `${balance.asset_code}:${balance.asset_issuer}`;
+  return "";
```

Comment:
i recognize that this got a little messy, but we have a lot of intersecting types now and this was the simplest way I could get this to compile without errors

Reply:
I know I'm just drive-by reviewing here, but what about

```ts
switch (balance.asset_type) {
  case "native":
    return balance.asset_type;

  case "credit_alphanum4":
  case "credit_alphanum12":
    return `${balance.asset_code}:${balance.asset_issuer}`;

  default:
    return "";
}
```

Reply:
What is the case that is failing here? I'm wondering when I'd expect to get `""`.

Reply:
Actually, the failing case here should be "native", because if there isn't `asset_code` or `asset_issuer`, it should assume "native"